app/models/auth/voms.rb

```diff
@@ -41,10 +41,21 @@ def parse_hash_groups!(hash)
         raise Errors::AuthenticationError, 'voms group env variable is not set' unless hash.key?('GRST_VOMS_FQANS')
         groups = Hash.new { |h, k| h[k] = [] }
         hash['GRST_VOMS_FQANS'].split(';').each do |line|
-          matches = line.match(VOMS_GROUP_REGEXP)
-          groups[matches[:group]] << matches[:role] if matches && matches[:role] != 'NULL'
+          group = parse_group!(line)
+          groups.merge!(group) { |_, oldval, newval| oldval + newval } if group
+        end
+        groups.map { |key, value| { id: key, roles: value.uniq } }
+      end
+
+      def parse_group!(line)
+        matches = line.match(VOMS_GROUP_REGEXP)
+        raise Errors::AuthenticationError, 'voms group env variable has invalid format' unless matches
+        return if matches[:group].include?('/')
```

Comment:
So just to be clear. We're raising an error if the line doesn't match the regexp and we're ignoring those lines that include subgroups (`/`). @arax is that the desired behaviour? I'm not against it I just want to be sure. Maybe we should at least log the match with the subgroup.

Reply:
Good point. `logger.warn` would be nice.